src/robot/model/body.py

```diff
@@ -57,15 +68,16 @@ def id(self):
             return 'k1'
         return self._get_id(self.parent)
 
-    def _get_id(self, parent):
+    def _get_id(self, parent: 'TestSuite|TestCase|BodyItem') -> str:
         steps = []
         if getattr(parent, 'has_setup', False):
-            steps.append(parent.setup)
+            steps.append(parent.setup)  # type: ignore # Use Protocol with RF 7
         if hasattr(parent, 'body'):
-            steps.extend(step for step in parent.body.flatten()
+            steps.extend(step for step in
+                         parent.body.flatten()  # type: ignore # Use Protocol with RF 7
                          if step.type != self.MESSAGE)
         if getattr(parent, 'has_teardown', False):
-            steps.append(parent.teardown)
+            steps.append(parent.teardown) # type: ignore # Use Protocol with RF 7
```

Comment:
PEP-8 says there should be at least two spaces before an inline comment, but we generally prefer four. I also believe the "second comment" mentioning Protocol should have four spaces before it as well to make it stand out. I can fix these myself.